1-Draft/RFCNNNN-Notification-On-Version-Update.md

```diff
@@ -241,6 +242,19 @@ This is comparatively the easy part.
 - If a file path is returned, then get the version information from the file name.
 - Use that version to construct the notification message, including the URL to that GitHub release page.
 
+#### How to control the update notification behavior using an environment variable
+
+The environment variable `POWERSHELL_UPDATECHECK` will be introduced to control the behavior of the update notification feature.
+The environment variable supports 3 values:
+
+- `Default`. This gives you the default behaviors:
+  - `pwsh` of preview versions check for the new preview version as well as the new GA version.
+  - `pwsh` of GA versions check for the new GA version only.
+
+- `Off`. This turns off the update notification feature.
+
+- `LTS`. `pwsh` of both preview and stable versions check for the new LTS GA version only.
```

Comment:
What if we is on preview,  set "Default" but want to continue to get preview notifications after GA?

Reply:
`LTS` should not be the default value. By default, a GA version should always notify the user about a new GA version.
Only if the user chooses to stay on `LTS`, then powershell should only report new `LTS` version.

I see your point, and we could add another configuration `All` (or a different name) so that even if you are on a GA version, you can still get notified about a new preview version.

Reply:
I think about this more, and find the `All` option (always detect a newer version, no matter the new version is preview or stable) not that useful. A preview version would come out about 4 weeks after the GA version gets released, so it's very possible one will use the GA with the preview side-by-side. By setting the `All` option, you will have the GA version notify "a new preview version" almost all the time except for the first 4 weeks, so basically you will have to change it to `Default` or `LTS` after you get the first "new preview version" notification. It sounds useful to powershell enthusiast (like you😄), but might not be that useful to most regular users. I will postpone supporting the `All` option until it has a higher demand.

Reply:
I agree.